portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -90,22 +93,44 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
 
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    // Take the input from the POST request 
-    String text = request.getParameter("text-input");
-    String author = request.getParameter("author");
-    // The added date for the comment will be the current date.
-    Date date = new Date();
-
-    Entity commentEntity = new Entity("Comment");
-    commentEntity.setProperty("text", text);
-    commentEntity.setProperty("author", author);
-    commentEntity.setProperty("date", date);
-
-    // Store comment data in the database
-    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
-    datastore.put(commentEntity);
+    // Check that the user is authenticated before adding the comment
+    UserService userService = UserServiceFactory.getUserService();
+
+    if (userService.isUserLoggedIn()) {
+      // Take the comment input from the POST request 
+      String text = request.getParameter("comment-text");
+      
+      String userId = userService.getCurrentUser().getUserId();
+      String email = userService.getCurrentUser().getEmail();
+
+      Entity commentEntity = new Entity("Comment");
+      commentEntity.setProperty("text", text);
+      commentEntity.setProperty("author", getUsername(userId));
+      commentEntity.setProperty("email", email);
+      // The added date for the comment will be the current date.
+      commentEntity.setProperty("date", new Date());
```

Comment:
I think it's OK to ignore here. But ideally, this should be timestamp so that you could localize to different timezone later when necessary.

Reply:
Noted, thank you for the remark!